aten/src/ATen/native/cuda/UpSampleNearest1d.cu

```diff
@@ -11,92 +11,82 @@ namespace at {
 namespace native {
 namespace {
 
-template <typename scalar_t, typename accscalar_t>
+#define MAX_THREADS 512
+
+template <typename scalar_t>
 C10_LAUNCH_BOUNDS_1(1024)
 __global__ void upsample_nearest1d_out_frame(
-    const int n,
-    const PackedTensorAccessor<scalar_t, 3> idata,
-    PackedTensorAccessor<scalar_t, 3> odata) {
-  int index = threadIdx.x + blockIdx.x * blockDim.x;
-
-  const int batchsize = idata.size(0);
-  const int channels = idata.size(1);
-  const int width1 = idata.size(2);
-  const int width2 = odata.size(2);
-
-  const float scale = (float)width1 / (float)width2;
-
-  if (index < n) {
-    const int w2 = index % width2;
-    // special case: just copy
-    if (width1 == width2) {
-      const int w1 = w2;
-      for (int n = 0; n < batchsize; n++) {
-        for (int c = 0; c < channels; ++c) {
-          const scalar_t val = idata[n][c][w1];
-          odata[n][c][w2] = val;
-        }
-      }
-      return;
-    }
-    //
-    const int w1 = nearest_neighbor_compute_source_index(scale, w2, width1);
-
-    for (int n = 0; n < batchsize; n++) {
-      for (int c = 0; c < channels; ++c) {
-        const scalar_t val = idata[n][c][w1];
-        odata[n][c][w2] = val;
-      }
-    }
+    const scalar_t* input,
+    size_t dim_b,
+    size_t dim_c,
+    size_t src_dim_w,
+    size_t dst_dim_w,
+    scalar_t* output) {
+  size_t dst_idx = blockIdx.x * blockDim.x + threadIdx.x;
+  if (dst_idx >= dim_c * dst_dim_w)
+    return;
+
+  float scale_factor = (float)dst_dim_w / src_dim_w;
+
+  int c = (dst_idx / dst_dim_w) % dim_c;
+
+  int dst_x = dst_idx % dst_dim_w;
+  int src_x = min((int)(dst_x / scale_factor), (int)src_dim_w - 1);
```

Comment:
Could you use `nearest_neighbor_compute_destination_index` here? Math seems the same. Even if you couldn't, `static_cast` instead of `(int)`

Reply:
Will switch to `nearest_neighbor_compute_source_index`.